Add wss example

kt3k opened the pull request:
This PR adds an example usage of this library, which shows how to create the client with `wss://` endpoint.

at-wat commented on .travis.yml:
```diff
@@ -12,6 +12,7 @@ env:
     - REVIEW_DOG_ON=1.11
     - REVIEW_DOG_VERSION=0.9.11
     - VERSION_DIR=v2
+    - PKGS="./ ./awsiotprotocol ./presigner ./pubqueue ./subqueue"
```
IMO, it's better to apply go vet and golint also to samples.

at-wat commented on v2/sample/wss/main.go:
```diff
@@ -0,0 +1,91 @@
+		OfflineQueueDropBehavior: "oldest",
+		AutoResubscribe:          true,
+		OnConnectionLost: func(opt *awsiot.Options, mqttOpt *mqtt.ClientOptions, err error) {
+			fmt.Printf("Connection lost handler function called\n")
```
Presigned URI should be regenerated at here.

at-wat commented on v2/sample/wss/main.go:
```diff
@@ -0,0 +1,91 @@
+// Copyright 2018 SEQSENSE, Inc.
```
would be nice to rename `sample` to `examples`, and put original example to an another directory (`mqtt`?).

kt3k commented:
@at-wat Thank you for the review. I updated the following points:

- Renamed `sample` to `example` and reorganized `example` dir.
- Added `$EXAMPLES` env var to lint and vet them in travis.
- Added signing code in OnConnectionLost hanlder (wss example)

kt3k commented:
followed v3 update

kt3k commented on .travis.yml:
```diff
@@ -28,7 +28,7 @@ script:
     echo "Enable coverage report"
     TEST_ARG="-coverprofile=coverage.txt -covermode=atomic -coverpkg=all"
-  - go test ${TEST_ARG} ./...
+  - go test ${TEST_ARG} ./ ./presigner ./internal/...
```
We can't include example packages in `go test` because it causes `flag redefined` panic. Probably that is a similar issue with this one https://github.com/mattn/goveralls/issues/89

at-wat replied:
Adding build tag to the files to be excluded would be cleaner.
```diff
diff --git a/example/mqtt/main.go b/example/mqtt/main.go
index b1ecbfe..b52e638 100644
--- a/example/mqtt/main.go
+++ b/example/mqtt/main.go
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// +build examples
+
 /**
   Usage: go run . -thing-name NAME \
                   -private-key-path path/to/private-key \
diff --git a/example/wss/main.go b/example/wss/main.go
index 54465a7..be1ed45 100644
--- a/example/wss/main.go
+++ b/example/wss/main.go
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// +build examples
+
 /**
   Usage: go run . -host hoge-ats.iot.ap-northeast-1.amazonaws.com
 */
```
`go test ./...` doesn't build these files without adding `-tags=examples`

at-wat replied:
Sorry, it's inconvenient to build example. (requires `go build . -tags=example`)

How about 
```
go test $(go list ./... | grep -v "/example/")
```
?
I think it would be better to test basically all files excepting some specific things.

at-wat commented:
I think `examples` is more common rather than `example`. Package name is preferred to be singular, but this directory is not a package.

at-wat commented on internal/awsiotprotocol/mqtts_test.go:
```diff
@@ -30,9 +30,9 @@ func TestMqttsNewClientOptions(t *testing.T) {
-		KeyPath:  "../../sample/samplecerts/client-key.pem",
+		CaPath:   "../../examples/mqtt/samplecerts/cafile.pem",
+		CertPath: "../../examples/mqtt/samplecerts/client-crt.pem",
+		KeyPath:  "../../examples/mqtt/samplecerts/client-key.pem",
```
Sorry, these files are used only by this test. `samplecerts` seems better to be moved to this directory.

kt3k replied:
OK. Makes sense.
